spring-cloud-gcp-pubsub-stream-binder/src/main/java/com/google/cloud/spring/stream/binder/pubsub/PubSubMessageChannelBinder.java

```diff
@@ -75,8 +76,15 @@ protected MessageHandler createProducerMessageHandler(
       ExtendedProducerProperties<PubSubProducerProperties> producerProperties,
       MessageChannel errorChannel) {
 
-    PubSubMessageHandler messageHandler =
-        new PubSubMessageHandler(this.pubSubTemplate, destination.getName());
+    PubSubMessageHandler messageHandler = new PubSubMessageHandler(this.pubSubTemplate, destination.getName());
+
+    String [] headers = producerProperties.getExtension().getAllowedHeaders();
```

Comment:
The next line checks that `producerProperties.getExtension()` is not null, but if it is null, then there will be a NullPointerException thrown on this line.